app/src/main/java/org/listenbrainz/android/viewmodel/UserViewModel.kt

```diff
@@ -71,13 +71,17 @@ class UserViewModel @Inject constructor(
         if(username.isNullOrEmpty()) return
         viewModelScope.launch (ioDispatcher) {
             socialRepository.followUser(username)
```

Comment:
Can we change the state instantly and revert back if something goes wrong? Here:
```kotlin
listenStateFlow.value = listenStateFlow.value.copy(isFollowing = true)
val result = socialRepository.followUser(username)
if (result.status == Resource.Status.FAILED) {
    listenStateFlow.value = listenStateFlow.value.copy(isFollowing = false)
}
```

This way, we would have quick response to user tap, but in reality, we would still be processing api request in the background. If the request fails, we simply reflect that to the UI at a later stage.